boards/nxp/frdm_mcxn236/board.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024  NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/clock/mcux_lpc_syscon_clock.h>
+#include <fsl_clock.h>
+#include <fsl_spc.h>
+#include <soc.h>
+
+/* Board xtal frequency in Hz */
+#define BOARD_XTAL0_CLK_HZ                        24000000U
+/* Core clock frequency: 150MHz */
+#define CLOCK_INIT_CORE_CLOCK                     150000000U
+/* System clock frequency. */
+extern uint32_t SystemCoreClock;
+
+__ramfunc static void enable_lpcac(void)
+{
+	SYSCON->LPCAC_CTRL |= SYSCON_LPCAC_CTRL_CLR_LPCAC_MASK;
+	SYSCON->LPCAC_CTRL &= ~(SYSCON_LPCAC_CTRL_CLR_LPCAC_MASK |
+				SYSCON_LPCAC_CTRL_DIS_LPCAC_MASK);
+}
+
+/* Update Active mode voltage for OverDrive mode. */
+void power_mode_od(void)
+{
+	/* Set the DCDC VDD regulator to 1.2 V voltage level */
+	spc_active_mode_dcdc_option_t opt = {
+		.DCDCVoltage       = kSPC_DCDC_OverdriveVoltage,
+		.DCDCDriveStrength = kSPC_DCDC_NormalDriveStrength,
+	};
+	SPC_SetActiveModeDCDCRegulatorConfig(SPC0, &opt);
+
+	/* Set the LDO_CORE VDD regulator to 1.2 V voltage level */
+	spc_active_mode_core_ldo_option_t ldo_opt = {
+		.CoreLDOVoltage       = kSPC_CoreLDO_OverDriveVoltage,
+		.CoreLDODriveStrength = kSPC_CoreLDO_NormalDriveStrength,
+	};
+	SPC_SetActiveModeCoreLDORegulatorConfig(SPC0, &ldo_opt);
+
+	/* Specifies the 1.2V operating voltage for the SRAM's read/write timing margin */
+	spc_sram_voltage_config_t cfg = {
+		.operateVoltage       = kSPC_sramOperateAt1P2V,
+		.requestVoltageUpdate = true,
+	};
+	SPC_SetSRAMOperateVoltage(SPC0, &cfg);
+}
+
+static int frdm_mcxn236_init(void)
+{
+	enable_lpcac();
+
+	power_mode_od();
+
+	/* Enable SCG clock */
+	CLOCK_EnableClock(kCLOCK_Scg);
+
+	/* FRO OSC setup - begin, enable the FRO for safety switching */
+
+	/* Switch to FRO 12M first to ensure we can change the clock setting */
+	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
+
+	/* Configure Flash wait-states to support 1.2V voltage level and 150000000Hz frequency */
+	FMU0->FCTRL = (FMU0->FCTRL & ~((uint32_t)FMU_FCTRL_RWSC_MASK)) | (FMU_FCTRL_RWSC(0x3U));
```

Comment:
Any reason not to use `CLOCK_SetFLASHAccessCyclesForFreq` here?

Reply:
This was copied from frdm_mcxn947 board.c.